### PR TITLE
add queries to GET /api/articles

### PR DESCRIPTION
**Pull request by catherine-newman**

### CODE REVIEW

File: __tests__/controllers.test.js [hunk 1]
```diff
@@ -99,6 +99,72 @@ describe("GET /api/articles", () => {
...
+        .expect(200);
+        const articles = res.body.articles;
+        expect(articles).toBeSortedBy("author", { descending : true });
+    })
+    test("articles can be sorted by votes", async () => {
+        const res = await request(app)
+        .get("/api/articles?sort_by=votes")
+        .expect(200);
+        const articles = res.body.articles;
+        expect(articles).toBeSortedBy("votes", { descending : true });
```
Comment (MKD01) [inline __tests__/controllers.test.js:121]: You dont have to do multiple tests for checking that the articles can be sorted by all of these, simply having one test that is not sorting by the default (which is by date) is enough to ensure that this works 👍

File: __tests__/controllers.test.js [hunk 2]
```diff
@@ -99,6 +99,72 @@ describe("GET /api/articles", () => {
...
+        .get("/api/articles?topic=mitch&sort_by=votes&order=asc")
+        .expect(200);
+        const articles = res.body.articles;
+        expect(articles).toHaveLength(12);
+        expect(articles).toBeSortedBy("votes", { descending : false });
+        articles.forEach(article => {
+            expect(article).toHaveProperty("topic", "mitch");
+        })
+
+    })
```
Comment (MKD01) [inline __tests__/controllers.test.js:167]: These are all really great tests, but there is one more test you can add to this. 

What should we expect to happen if a user queries by a topic that exists but has no articles associated to it?

**Review by MKD01 — CHANGES_REQUESTED** [review]
Great work so far, but I would like to see that last test added before you merge, so once you have added that please send this back as a new pr.

**Review by yalicee — APPROVED** [review]
Looks good :)

**Pull request merged**